### PR TITLE
MGMT-17963: Return clearer error messages for image registry

### PR DESCRIPTION
**Pull request by CrystalChun**
Creates and returns error messages that indicate where the image registry creation failed.

Includes small refactoring of creating the registry configmap and test cleanup.

### CODE REVIEW

**Comment by rccrdpccl** [comment]
sorry for causing the conflict @CrystalChun , just wanted to make sure we add linter to the process

**Comment by CrystalChun** [comment]
> sorry for causing the conflict @CrystalChun , just wanted to make sure we add linter to the process

No problem! What do you think of the error messages?

File: controlplane/internal/imageregistry/image_registry_test.go [hunk 1]
```diff
@@ -217,13 +226,13 @@ var _ = Describe("ImageRegistry Test", func() {
...
-				imageRegistryConfigMap, err := CreateImageRegistryConfigmap(
+				By("Calling the GenerateImageRegistryConfigmap function")
+				imageRegistryConfigMap, err := GenerateImageRegistryConfigmap(
 					userRegistryCM,
-					namespace,
+					testNamespace,
 				)
 				Expect(err).To(HaveOccurred())
-				Expect(err.Error()).To(Equal("(6, 13): cannot have two dots in one float"))
+				Expect(err.Error()).To(ContainSubstring("(6, 13): cannot have two dots in one float"))
```
Comment (rccrdpccl) [inline controlplane/internal/imageregistry/image_registry_test.go:235]: Can we see an example of descriptive error message here? Would this be `failed to parse image mirror registry toml for registry <name> : (6, 13): cannot have two dots in one float`?

File: controlplane/internal/imageregistry/image_registry.go [hunk 2]
```diff
@@ -142,28 +148,30 @@ func parseMirrorRegistries(
...
+				itmsMirror.Source = source
+				itmsMirror.Mirrors = append(itmsMirror.Mirrors, configv1.ImageMirror(mirror))
+			case "digest-only":
 				idmsMirror.Source = source
 				idmsMirror.Mirrors = append(idmsMirror.Mirrors, configv1.ImageMirror(mirror))
 			}
+		} else {
+			// Default is pulling by digest
+			idmsMirror.Source = source
+			idmsMirror.Mirrors = append(idmsMirror.Mirrors, configv1.ImageMirror(mirror))
```
Comment (rccrdpccl) [inline controlplane/internal/imageregistry/image_registry.go:174]: Can we extract this to a function? Doing so we'll also get rid of the `else` condition.

Comment (CrystalChun) [reply controlplane/internal/imageregistry/image_registry.go]: Modified, how does it look?

File: controlplane/internal/imageregistry/image_registry_test.go (hunk 1 above)
Comment (CrystalChun) [reply controlplane/internal/imageregistry/image_registry_test.go]: Yeah! Edited to match the error message

File: controlplane/internal/imageregistry/image_registry_test.go [hunk 3]
```diff
@@ -217,13 +226,14 @@ var _ = Describe("ImageRegistry Test", func() {
...
+				By("Calling the GenerateImageRegistryConfigmap function")
+				imageRegistryConfigMap, err := GenerateImageRegistryConfigmap(
 					userRegistryCM,
-					namespace,
+					testNamespace,
 				)
 				Expect(err).To(HaveOccurred())
-				Expect(err.Error()).To(Equal("(6, 13): cannot have two dots in one float"))
+				Expect(err.Error()).To(Equal(
+					"failed to load value of registries.conf into toml tree; incorrectly formatted toml: (6, 13): cannot have two dots in one float"))
```
Comment (rccrdpccl) [inline controlplane/internal/imageregistry/image_registry_test.go:236]: much clearer :)

**Comment by CrystalChun** [comment]
Will be included in https://github.com/openshift-assisted/cluster-api-agent/pull/41

**Pull request closed without merging**